Ruler: Add pagination for api/v1/rules (#9563)
grafana/mimir: docs/sources/mimir/references/http-api/index.md

The `exclude_alerts` parameter is optional. If set, it only returns rules and excludes active alerts.

> The `group_limit` and `group_next_token` parameters are optional. If `group_limit` is set, it will limit the number of rule groups returned in a single response. If the total number of rule groups exceeds this value, the response will contain a `groupNextToken`.

Reviewer: ```suggestion
The `group_limit` and `group_next_token` parameters are optional. If `group_limit` is set, it limits the number of rule groups returned in a single response. If the total number of rule groups exceeds this value, the response contains a value of `groupNextToken`.
```

Reviewer: We avoid using "will" in docs in favor of present tense.

Reviewer 2: Should these two new parameters be added to the endpoint definition itself? I.e., `GET <prometheus-http-prefix>/api/v1/rules?type={alert|record}&file={}&rule_group={}&rule_name={}&exclude_alerts={true|false}`.

> This can be passed into subsequent requests via `group_next_token` to paginate over the remaining groups. The final response will not contain a token.

Reviewer: ```suggestion
You can pass this value into subsequent requests via the `group_next_token` parameter to paginate over the remaining groups. The final response doesn't contain a token.
```
